### PR TITLE
plugin-space: fix duplicate settings space created during onboarding

### PR DESCRIPTION
**Pull request by wittjosiah**
## Problem

Every fresh onboarding created **two** `org.dxos.space.settings`-tagged spaces instead of one (4 spaces total instead of 3). Two independent creators race, and the race is guaranteed to be lost:

1. `identity-created.ts` runs `AppSpace.setupIdentitySpaces`, which deliberately creates the default space first (so it is first in `client.spaces.get()`), awaits `waitUntilReady()` + edge-replication setup, and only then creates the settings space and stamps the `defaultSpace` annotation.
2. `spaces-ready.ts` forks its settings-space init the moment `client.spaces.get().length > 0` — squarely inside that awaited window — and `ensureSettingsSpace` finds no tagged space yet, so it creates the duplicate. The migration then puts the shared-spaces ordering Expando in it.

Secondary damage: the duplicate (annotation-less) space is created first, so `AppSpace.getSettingsSpace` (a `.find()` over list order) returned it and `getDefaultSpace()` missed the designation.

## Fix

- **`plugin-space/util/settings-space.ts`** — new `resolveSettingsSpace`: creation from the spaces-ready path now happens **only** for legacy profiles that predate the settings space (the reason this path could create at all). Otherwise it subscribes to `client.spaces` and waits for the tagged space, which is guaranteed to arrive — genesis creates it for fresh profiles, replication delivers it for existing profiles on a new device. The check runs inside the subscription callback (which replays current state), so there is no gap in which an arriving settings space could be missed. This removes the second creator outright rather than shrinking a check-then-create window.
- **`spaces-ready.ts`** — `initSettingsSpace` uses `resolveSettingsSpace` instead of `ensureSettingsSpace`.
- **`app-toolkit/AppSpace.ts`** — `getSettingsSpace` now prefers the tagged space that is ready and carries the default-space designation, falling back to list order. This repairs profiles that already carry a duplicate: `getDefaultSpace()` resolves the canonical space again, at the cost of the (near-empty) ordering Expando in the duplicate being ignored and recreated in the canonical space. The stray duplicate space itself is not deleted — it is hidden, locked, and inert; a full merge/delete was judged not worth the risk given the bug shipped recently and affected profiles are fresh onboardings.

## Tests

New `settings-space.test.ts` pins the genesis interleaving with a real client: a resolver fiber forked *before* `setupIdentitySpaces` (strictly wider than the real race window) must resolve the genesis-created space, the profile must end with exactly one settings space holding **both** the ordering Expando and the designation, and `getDefaultSpace()` must resolve the default space. Two more tests cover the legacy-profile create path and duplicate-profile canonical resolution.

```
plugin-space:  ✓ 16 files passed | 1 skipped — 50 tests passed | 3 todo
app-toolkit:   ✓ 13 files passed — 142 tests passed
lint: clean; formatted with oxfmt
```

Manual verification against a local edge stack (fresh onboarding → 3 spaces, `getDefaultSpace()` → "My Space", MCP consent page lists 3 space ids) still needs a human run — this environment has no local stack.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_014dfiDoDAGmswUzTbQQb885

---
_Generated by [Claude Code](https://claude.ai/code/session_014dfiDoDAGmswUzTbQQb885)_

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented duplicate settings spaces from being created during onboarding.
  * Improved settings-space selection when duplicates exist by favoring the designated default.
  * Added support for legacy profiles without an existing settings space.
  * Ensured settings spaces are ready before they’re used, improving reliability during startup.

### CODE REVIEW

File: packages/plugins/plugin-space/src/util/settings-space.ts [hunk 1]
```diff
@@ -6,14 +6,47 @@ import * as Effect from 'effect/Effect';
...
+    const sub = client.spaces.subscribe(() => {
+      const settingsSpace = AppSpace.getSettingsSpace(client);
+      if (settingsSpace) {
+        resume(Effect.succeed(settingsSpace));
+      } else if (AppSpace.resolveLegacyDefaultSpace(client)) {
+        resume(Effect.succeed(undefined));
+      }
+    });
+    return Effect.sync(() => sub.unsubscribe());
+  });
```
[the remark this answers is not in the document]
Comment (wittjosiah) [reply packages/plugins/plugin-space/src/util/settings-space.ts:35]: Verified against the code — the window exists but is self-correcting, so I'm not taking the suggested change.

If the resolver completes with the undesignated duplicate (only possible on a profile that already carries duplicates), the only thing bootstrap does with it is `migrateToSettingsSpace`, which on this path just re-ensures the ordering Expando the duplicate already holds — the designation is never written or moved (no legacy space on fresh-dupe profiles). `resolveDefaultSpace` then loops, and each iteration re-runs `getSettingsSpace` via `getDefaultSpace(client)`; the canonical space becoming ready fires the space-list subscription the loop awaits, so it converges on the designated space and exits correctly. Nothing is finalized against the wrong space, and all other consumers call `getSettingsSpace` live rather than caching the resolver's result.

Deferring completion until canonical selection is "stable" would also make bootstrap hang on any profile whose designated duplicate never opens — the fallback is deliberate boot resilience for an already-degraded state.

---
_Generated by [Claude Code](https://claude.ai/code)_
